src/index.ts

```diff
@@ -224,6 +224,45 @@ function b32decodeXemAddr(data: string): Buffer {
   return nemSdk.default.utils.convert.ua2hex(nemSdk.default.model.address.b32decode(address));
 }
 
+function ontBase58CheckEncoder(addrVersion: number): (data: Buffer) => string {
+  return (data: Buffer) => {
+   let addr: Buffer;
+   switch(addrVersion) {
```

Comment:
Can you explain what's going on here? Why add a version byte to the decoded data? Is ONT expecting multiple address types? Where does the value 0x17 come from?

Reply:
ADDR_VERSION in https://github.com/ontio/ontology-ts-sdk/blob/master/src/consts.ts was changed once "// export const ADDR_VERSION = '41';
export const ADDR_VERSION = '17';
" maybe not the last time.

Reply:
Мaybe in the future it can be used for other coins

Reply:
But why 17? And why have a prefix in the decoded data at all?

Reply:
Q: Why have a prefix in the decoded data at all?
A: Above you asked:

![Screenshot_1](https://user-images.githubusercontent.com/12184447/70122065-96cb1680-1678-11ea-92c1-3aa62ee95470.png)

Q: Why 17?
A:
This is original function:
https://github.com/ontio/ontology-ts-sdk/blob/master/src/crypto/address.ts:
function hexToBase58(hexEncoded: string): string {
    const data = ADDR_VERSION + hexEncoded;

    const hash = sha256(data);
    const hash2 = sha256(hash);
    const checksum = hash2.slice(0, 8);

    const datas = data + checksum;

    return base58.encode(new Buffer(datas, 'hex'));
}
where ADDR_VERSION = '17' in hex, because "base58.encode" takes parameter "new Buffer(datas, 'hex')".
If don’t need to pass the parameter, then I can write this function like this:
function ontBase58CheckEncoder(data: Buffer): string {
  let addr: Buffer;
  const ADDR_VERSION = '17';
  if(data.length !== 20) {
    throw Error('Unrecognised address format');
  }
  addr = Buffer.concat([new Buffer(ADDR_VERSION,'hex'), data]);
  return bs58check.encode(addr);
}

Reply:
Tell me what should i do?

Reply:
Answer me please.

Reply:
Ontology and NEO addresses are the same, this function can be used for both of them.